python/doc/source/plasma.rst

```diff
@@ -0,0 +1,680 @@
+``pyarrow.write_tensor``:
+
+.. code-block:: python
+
+  # Write the tensor into the Plasma-allocated buffer
+  stream = pa.FixedSizeBufferOutputStream(buf)
+  pa.write_tensor(tensor, stream)  # Writes tensor's 552 bytes to Plasma stream
+
+To finish storing the Arrow object in Plasma, you can seal it just like 
+for any other data:
+
+.. code-block:: python
+
+  # Seal the Plasma object
+  client.seal(object_id)
+
+Getting Arrow Objects from Plasma
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For reading the object from Plasma to Arrow, you can fetch it as a ``PlasmaBuffer`` 
+using its object id as usual. 
+
+.. code-block:: python
+
+  # Get the arrow object by ObjectID.
+  [buf2] = client.get([object_id])
+
+To convert the ``PlasmaBuffer`` back into the Arrow ``Tensor``, first you have to 
+create a pyarrow ``BufferReader`` object from it. You can then pass the 
+``BufferReader`` into ``pyarrow.read_tensor`` to reconstruct the Arrow ``Tensor``
+object:
+
+.. code-block:: python
+
+  # Reconstruct the Arrow tensor object.
+  reader = pa.BufferReader(buf2)  # Plasma buffer -> Arrow reader
+  tensor2 = pa.read_tensor(reader)  # Arrow reader -> Arrow tensor
+
+Finally, you can use ``pyarrow.read_tensor`` to convert the Arrow object 
+back into numpy data:
+
+.. code-block:: python
+
+  # Convert back to numpy
+  array = tensor2.to_numpy()  # Arrow tensor -> numpy array
+
+Storing Pandas DataFrames in Plasma
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Storing a Pandas ``DataFrame`` still follows the *create* then *seal* process 
+of storing an object in the Plasma store, however one cannot directly write 
+the ``DataFrame`` to Plasma with Pandas alone. Plasma also needs to know the 
+size of the ``DataFrame`` to allocate a buffer for. 
+
+One can instead use pyarrow and its supportive API as an intermediary step 
+to import the Pandas ``DataFrame`` into Plasma. Arrow has multiple equivalent 
+types to the various Pandas structures, see the :ref:`pandas` page for more.
+
+You can create the pyarrow equivalent of a Pandas ``DataFrame`` by using 
+``pyarrow.from_pandas`` to convert it to a ``RecordBatch``.
+
+.. code-block:: python
+
+  import pyarrow as pa
+  import pandas as pd
+
+  # Create a Pandas DataFrame
+  d = {'one' : pd.Series([1., 2., 3.], index=['a', 'b', 'c']),
+       'two' : pd.Series([1., 2., 3., 4.], index=['a', 'b', 'c', 'd'])}
+  df = pd.DataFrame(d)
+
+  # Convert the Pandas DataFrame into a PyArrow RecordBatch
+  record_batch = pa.RecordBatch.from_pandas(df)
+
+Creating the Plasma object requires an ``ObjectID`` and the size of the
+data. Now that we have converted the Pandas ``DataFrame`` into a PyArrow 
+``RecordBatch``, use the ``MockOutputStream`` to determine the
+size of the Plasma object.
+
+.. code-block:: python
+
+  # Create the Plasma object from the PyArrow RecordBatch
+  object_id = plasma.ObjectID(np.random.bytes(20))
+  mock_sink = pa.MockOutputStream()
```

Comment:
Ideally we would use get_record_batch_size here, but at the moment it returns the wrong size. This should be fixed before this PR gets merged.